src/components/molecules/token-input/token-input.tsx

```diff
@@ -59,7 +59,7 @@ export default function TokenInput(props: TokenInputProps) {
               handleChange(tokenValue)
             }}
         />
-        {!disabled &&
+        {(!disabled && balance) &&
```

Comment:
```suggestion
        {(!disabled && balance !== undefined) &&
```
So that the balance is still displayed if happens to be 0.